test/cri-containerd/stats_test.go

```diff
@@ -354,3 +365,76 @@ func Test_ContainerStats_List_ContainerID(t *testing.T) {
 		})
 	}
 }
+
+func Test_SandboxStats_WorkingSet_PhysicallyBacked(t *testing.T) {
+	type config struct {
+		name             string
+		requiredFeatures []string
+		runtimeHandler   string
+		sandboxImage     string
+	}
+	tests := []config{
+		{
+			name:             "WCOW_Hypervisor",
+			requiredFeatures: []string{featureWCOWHypervisor},
+			runtimeHandler:   wcowHypervisorRuntimeHandler,
+			sandboxImage:     imageWindowsNanoserver,
+		},
+		{
+			name:             "LCOW",
+			requiredFeatures: []string{featureLCOW},
+			runtimeHandler:   lcowRuntimeHandler,
+			sandboxImage:     imageLcowK8sPause,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			requireFeatures(t, test.requiredFeatures...)
+
+			if test.runtimeHandler == lcowRuntimeHandler {
+				pullRequiredLcowImages(t, []string{test.sandboxImage})
+			} else {
+				pullRequiredImages(t, []string{test.sandboxImage})
+			}
+
+			var sizeInMB uint64 = 2048
+			sizeInMBStr := strconv.FormatUint(sizeInMB, 10)
+			podRequest := &runtime.RunPodSandboxRequest{
+				Config: &runtime.PodSandboxConfig{
+					Metadata: &runtime.PodSandboxMetadata{
+						Name:      t.Name(),
+						Uid:       "0",
+						Namespace: testNamespace,
+					},
+					Annotations: map[string]string{
+						"io.microsoft.virtualmachine.computetopology.memory.allowovercommit":      "false",
+						"io.microsoft.virtualmachine.computetopology.memory.enabledeferredcommit": "false",
+						"io.microsoft.virtualmachine.computetopology.memory.sizeinmb":             sizeInMBStr,
```

Comment:
This annotation just determines the max size in mb that a uvm can have, right? how are you making sure that you hit that size to guarantee the working set size is equivalent?

Reply:
I'm pretty sure it determines the amount of memory assigned to the UVM i.e to override the default of 1GB. When I was doing tests for this 1024, 2048 (here), and 3072 produced 1GB, 2Gb and 3GB working set as expected.